src/charm.py

```diff
@@ -121,8 +122,13 @@ def _on_update_runner_bin(self, event):
         if not runner_manager:
             return
         old_status = self.unit.status
-        self.unit.status = MaintenanceStatus("Checking for runner updates")
-        runner_bin_url = runner_manager.get_latest_runner_bin_url()
+        try:
+            self.unit.status = MaintenanceStatus("Checking for runner updates")
+            runner_bin_url = runner_manager.get_latest_runner_bin_url()
+        except urllib.error.URLError as e:
+            logger.exception("Failed to check for runner updates")
+            self.unit.status = BlockedStatus(f"Failed to check for runner updates: {e}")
```

Comment:
Do we really want to block here? Was if it's a transient error?

Reply:
Maybe it is better to switch to Maintenance status, as update binary is being trigger on a schedule (default to 1 hour).
However, the default one hour retry is a bit long, so block status may be a good choice.

Currently, failure to get the URL in `_on_install` method also return blocked status. So that should be changed as well, if we are changing to maintenance status.

Reply:
Yeah, I'd agree we should leave this as `BlockedStatus` for now and re-assess switching this to something else later (and updating all other entries in this charm that do something similar at the same time).